packages/@storefront/flux-capacitor/test/unit/core/store/middleware.ts

```diff
@@ -427,6 +432,48 @@ suite('Middleware', ({ expect, spy, stub }) => {
     });
   });
 
+  describe('redirectAnalyzer()', () => {
+    let next;
+
+    beforeEach(() => {
+      next = sinon.spy();
+    });
+
+    it('should invoke the next middleware by default', () => {
+      const action = { type: 'FOO' };
+
+      redirectAnalyzer()(next)(action);
+
+      expect(next).to.be.calledWithExactly(action);
+    });
+
+    it('should not invoke the next middleware if the action is of type START_REDIRECT', () => {
+      const action = { type: 'START_REDIRECT' };
+
+      const result = redirectAnalyzer()(next)(action);
+
+      expect(result).to.be.undefined;
+      expect(next).not.to.be.called;
+    });
+
+    it('should not invoke the next middleware if the START_REDIRECT action has been dispatched previously', () => {
```

Comment:
Do you think it's worth it to test the complement of this (i.e. after a `DONE_REDIRECT`, make sure that subsequent actions pass through)?

Reply:
@jmbtutor 

I actually went back-and-forth on this. My current thinking (and the reason that the corresponding test was omitted) is that the sole `DONE_REDIRECT` test actually accounts for this. Put another way, it doesn't actually matter if `START_REDIRECT` was dispatched prior to `DONE_REDIRECT`, so long as the latter ensures that subsequent actions are handled as normal.

In the spirit of 'completeness', I'm happy to add this test.

Reply:
An implementation that calls `next` on `DONE_REDIRECT` but not set `isRedirecting` (preventing future actions from passing through) will still pass these tests, which is why I wondered if we should have it. Perhaps we can merge this new test with the existing `DONE_REDIRECT` test, since it'll only pass if `DONE_REDIRECT` works properly.

If we do that, perhaps the sole `START_REDIRECT` test is redundant for the same reason.

Reply:
Those are good points. I'll consolidate the existing tests.